### ISSUE TITLE
Workaround for Podman problems with inconsistent Podman DB

### ISSUE THREAD

**Issue by krucod3** [eclipse-ankaios/ankaios#117]
As found by @maturar, Podman 3.4 returns error about an inconsistent state when executing `podman ps` shortly after a workload is scheduled. 
To retain the support for Ubuntu 20.04LTS and 22.04LTS (where we only get the old Podman 3.4 version) we need a workaround for that in order to not show `unknown` state for workloads that are already running and have no problems.

## Current Behavior
System shows unknow state for running workloads if Podman returns an error from the DB

## Expected Behavior
No unknown state is shown for running workloads.

## Steps to Reproduce

1. Start the ank-server with the startup configuration. You can use the configuration from the repository, but without hello-pod workload.
2. Start agents
3. Wait for a while and call ank get workloads
4. Delete all workloads step by step.
5. Stop the agents and the server.
6. Repeat previous steps until you see the error. 

## Logs

Here are traces from Ubuntu 20 (amd64) with the issue:
```
[2023-11-29T10:31:18Z DEBUG ank_agent::runtime_connectors::podman::podman_runtime] The workload 'hello2' has been created with id '32af8436ea999e6d61d95ca7080bc9a9723857f637e689632203509fc9d3fbbc'
[2023-11-29T10:31:18Z DEBUG ank_agent::runtime_connectors::podman::podman_runtime] Starting the checker for the workload 'hello2' with id '32af8436ea999e6d61d95ca7080bc9a9723857f637e689632203509fc9d3fbbc'
[2023-11-29T10:31:18Z TRACE ank_agent::runtime_connectors::podman::podman_runtime] Getting the state for the workload '32af8436ea999e6d61d95ca7080bc9a9723857f637e689632203509fc9d3fbbc'
...
[2023-11-29T10:31:18Z WARN  ank_agent::runtime_connectors::podman::podman_runtime] Could not get state of workload 'PodmanWorkloadId { id: "32af8436ea999e6d61d95ca7080bc9a9723857f637e689632203509fc9d3fbbc" }': 'Execution of command failed: Error: container 83db3d688cf5aa825f75ce0c6c514673af66664fd1b50c9f1d907f84888f8b1a does not exist in database: no such container
    '. Returning unknown.
[2023-11-29T10:31:18Z TRACE ank_agent::runtime_connectors::podman::podman_runtime] Returning the state Unknown
...
[2023-11-29T10:31:19Z TRACE ank_agent::runtime_connectors::podman::podman_runtime] Getting the state for the workload '32af8436ea999e6d61d95ca7080bc9a9723857f637e689632203509fc9d3fbbc'
[2023-11-29T10:31:19Z TRACE ank_agent::runtime_connectors::podman::podman_runtime] Returning the state Succeeded
```
## Final result

As a workaround we can ignore the error and execute an additional `podman ps` to get the state of the workloads. We should not just ignore the error as this could introduce some delay in the state handling. An extra query is executed in case of an error.

**Comment by maturar** [comment]
**Tasks:**
- [ ] Double-check the issue on target Ubuntu 20 and Ubuntu 22 on arm64. Get traces.
- [ ] Prepare the fix and test it.
- [ ] Find the bug tickets in Podman and link them with this ticket.

**Comment by maturar** [comment]
[Issue-117-logs.zip](https://github.com/eclipse-ankaios/ankaios/files/13559575/Issue-117-logs.zip)
Here are traces from Ankaios without the fix (taken from the main before I have made the fix).

**Comment by maturar** [comment]
I am testing the fix on Ubuntu 20 and Ubuntu 22, amd64. I can see that the `podman_runtime` still reports `Unknown` state for a workload from time to time. For example:
```
[2023-12-05T12:48:29Z DEBUG ank_agent::runtime_connectors::podman::podman_runtime] The workload 'nginx' has been created with id '6de05c1f77c1c3ce804788bb622e02df901fbb38feb42b3b312351b7c952f01d'
[2023-12-05T12:48:29Z DEBUG ank_agent::runtime_connectors::podman::podman_runtime] Starting the checker for the workload 'nginx' with id '6de05c1f77c1c3ce804788bb622e02df901fbb38feb42b3b312351b7c952f01d'
[2023-12-05T12:48:29Z TRACE ank_agent::runtime_connectors::podman::podman_runtime] Getting the state for the workload '6de05c1f77c1c3ce804788bb622e02df901fbb38feb42b3b312351b7c952f01d'
[2023-12-05T12:48:29Z TRACE ank_agent::runtime_connectors::podman::podman_runtime] Returning the state Unknown
....
[2023-12-05T12:48:30Z DEBUG ank_agent::generic_polling_state_checker] The workload nginx has changed its state to ExecRunning
```
But this seems to be a case, when the Podman really returns us the state `Unknown`. I do not see the warning trace here. The root cause is probably that the `nginx` has been starting and meanwhile from another thread the `podman_runtime` tried to get state.

**Comment by maturar** [comment]
The fix has been implemented. The retry for `podman ps` helps as a workaround. The issue described in the previous comment will be solved here: https://github.com/eclipse-ankaios/ankaios/issues/123.
Unfortunately I was not able to find a ticket when the Podman fixed the bug described here. I found out the issue https://github.com/containers/podman/issues/10120. But the fix from the issue has been released. With Podman 3.2. The tests I made here show, what the fix has been released somewhere between versions 3.4.2 and 4.7.2.
